zetaclient/inbound_tracker.go

```diff
@@ -0,0 +1,263 @@
+	"golang.org/x/net/context"
+)
+
+// ExternalChainWatcherForNewInboundTrackerSuggestions At each tick, gets a list of Inbound tracker suggestions from zeta-core and tries to check if the in-tx was confirmed.
+// If it was, it tries to broadcast the confirmation vote. If this zeta client has previously broadcast the vote, the tx would be rejected
+func (ob *EVMChainClient) ExternalChainWatcherForNewInboundTrackerSuggestions() {
+	ticker := NewDynamicTicker(fmt.Sprintf("EVM_ExternalChainWatcher_InboundTrackerSuggestions_%d", ob.chain.ChainId), ob.GetCoreParams().InTxTicker)
+	defer ticker.Stop()
+	ob.logger.ExternalChainWatcher.Info().Msg("ExternalChainWatcher for inboundTrackerSuggestions started")
+	for {
+		select {
+		case <-ticker.C():
+			err := ob.ObserveTrackerSuggestions()
+			if err != nil {
+				ob.logger.ExternalChainWatcher.Err(err).Msg("ObserveTrackerSuggestions error")
+			}
+			ticker.UpdateInterval(ob.GetCoreParams().InTxTicker, ob.logger.ExternalChainWatcher)
+		case <-ob.stop:
+			ob.logger.ExternalChainWatcher.Info().Msg("ExternalChainWatcher for inboundTrackerSuggestions stopped")
+			return
+		}
+	}
+}
+
+func (ob *BitcoinChainClient) ExternalChainWatcherForNewInboundTrackerSuggestions() {
+	ticker := NewDynamicTicker("Bitcoin_WatchInTx_InboundTrackerSuggestions", ob.GetCoreParams().InTxTicker)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C():
+			err := ob.ObserveTrackerSuggestions()
+			if err != nil {
+				ob.logger.WatchInTx.Error().Err(err).Msg("error observing in tx")
+			}
+			ticker.UpdateInterval(ob.GetCoreParams().InTxTicker, ob.logger.WatchInTx)
+		case <-ob.stop:
+			ob.logger.WatchInTx.Info().Msg("ExternalChainWatcher for BTC inboundTrackerSuggestions stopped")
+			return
+		}
+	}
+}
+
+func (ob *BitcoinChainClient) ObserveTrackerSuggestions() error {
+	trackers, err := ob.zetaClient.GetInboundTrackersForChain(ob.chain.ChainId)
+	if err != nil {
+		return err
+	}
+	for _, tracker := range trackers {
+		ob.logger.WatchInTx.Info().Msgf("checking tracker with hash :%s and coin-type :%s ", tracker.TxHash, tracker.CoinType)
+		ballotIdentifier, err := ob.CheckReceiptForBtcTxHash(tracker.TxHash, true)
+		if err != nil {
+			return err
+		}
+		ob.logger.WatchInTx.Info().Msgf("Vote submitted for inbound Tracker,Chain : %s,Ballot Identifier : %s, coin-type %s", ob.chain.ChainName, ballotIdentifier, common.CoinType_Gas.String())
+	}
+	return nil
+}
+
+func (ob *BitcoinChainClient) CheckReceiptForBtcTxHash(txHash string, vote bool) (string, error) {
+	hash, err := chainhash.NewHashFromStr(txHash)
+	if err != nil {
+		return "", err
+	}
+	tx, err := ob.rpcClient.GetRawTransactionVerbose(hash)
+	if err != nil {
+		return "", err
+	}
+	blockHash, err := chainhash.NewHashFromStr(tx.BlockHash)
+	if err != nil {
+		return "", err
+	}
+	block, err := ob.rpcClient.GetBlockVerbose(blockHash)
+	if err != nil {
+		return "", err
+	}
+	tss, err := ob.zetaClient.GetBtcTssAddress()
```

Comment:
you can use `ob.Tss.BTCAddress()` to avoid RPC query.

Reply:
i would like to keep it to the RPC query for now, to avoid any edge cases when the tss address changes. The one fetched from core would be correct at any time . the tss.CurrentPubkey might take block to be updated